src/Analyzers/CSharp/Tests/MakeDeclarationPartial/MakeDeclarationPartialTests.cs

```diff
@@ -0,0 +1,257 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.MakeDeclarationPartial;
+using Microsoft.CodeAnalysis.Editor.UnitTests.CodeActions;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.CSharp.UnitTests.MakeDeclarationPartial
+{
+    using VerifyCS = CSharpCodeFixVerifier<
+        EmptyDiagnosticAnalyzer,
+        CSharpMakeDeclarationPartialCodeFixProvider>;
+
+    [Trait(Traits.Feature, Traits.Features.CodeActionsMakeDeclarationPartial)]
+    public sealed class MakeDeclarationPartialTests
+    {
+        public static IEnumerable<object[]> AllValidDeclarationTypes()
+        {
+            yield return new[] { "class" };
+            yield return new[] { "struct" };
+            yield return new[] { "interface" };
+            yield return new[] { "record" };
+            yield return new[] { "record struct" };
+        }
+
+        [Theory]
+        [MemberData(nameof(AllValidDeclarationTypes))]
+        public async Task OutsideNamespace(string declarationType)
+        {
+            await new VerifyCS.Test
+            {
+                TestCode = $$"""
+                partial {{declarationType}} Declaration
+                {
+                }
+                
+                {{declarationType}} {|CS0260:Declaration|}
+                {
+                }
+                """,
+                FixedCode = $$"""
+                partial {{declarationType}} Declaration
+                {
+                }
+                
+                partial {{declarationType}} Declaration
+                {
+                }
+                """,
+                LanguageVersion = LanguageVersion.Preview
```

Comment:
Why do we need preview?

Reply:
The default language version was C#8 and it was not enough for records, so I've elevated it. Is anything particularly bad about preview? I mean, this fix should work regardless of version as soon as it supports all required language features, right?

Reply:
I personally would have used C# 10.